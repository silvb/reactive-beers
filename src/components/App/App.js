import React from 'react'
import Header from '../Header/Header'
import SearchArea from '../SearchArea/SearchArea'
import Dashboard from '../Dashboard/Dashboard'

const App = () => (
  <div className="app-container">
    <Header />
    <SearchArea />
    <Dashboard />
  </div>
)

export default App