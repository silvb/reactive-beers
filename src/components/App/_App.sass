.app-container
  width: 100%
  min-height: 100vh
  height: auto
  display: flex
  flex-flow: column nowrap
  align-items: stretch