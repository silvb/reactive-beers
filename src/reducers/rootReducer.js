const initialState = {
  beers: [],
  searchText: '',
  isFetching: false,
}

const rootReducer = (state = initialState, action) => {
  const actions = {
    UPDATE_SEARCH_TEXT: () => ({
      ...state,
      searchText: action.text,
    }),
    ORDER_BEERS: () => ({
      ...state,
      isFetching: true,
    }),
    RECEIVE_BEERS: () => ({
      ...state,
      
      isFetching: false,
      beers: action.beers
    }),
    default: () => state,
  }

  return (actions[action.type] || actions.default)()
}

export default rootReducer