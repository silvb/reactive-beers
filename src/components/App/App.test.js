import React from 'react'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import rootReducer from '../../reducers/rootReducer'
import renderer from 'react-test-renderer'
import App from './App'

describe('App', () => {
  it('renders without crashing', () => {
    const store = createStore(rootReducer)
    const component = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})