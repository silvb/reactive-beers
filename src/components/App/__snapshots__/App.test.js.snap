// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div
  className="app-container"
>
  <header>
    <h1>
      Reactive Beers 🍺🍺🍺
    </h1>
  </header>
  <div
    className="search-area"
  >
    <input
      className="search-area__input"
      onChange={[Function]}
      type="text"
      value=""
    />
    <div
      className="search-area__button-clear"
      onClick={[Function]}
    >
      Clear
    </div>
  </div>
  <div
    className="beercards"
  />
</div>
`;
